packages/framework/aqueduct/src/container-runtime-factories/baseContainerRuntimeFactory.ts

```diff
@@ -97,18 +101,64 @@ export class BaseContainerRuntimeFactory
 		this.registry = new FluidDataStoreRegistry(this.registryEntries);
 	}
 
-	public async instantiateFirstTime(runtime: ContainerRuntime): Promise<void> {
+	/**
+	 * @deprecated Use the overload taking in an IContainerRuntime
+	 */
+	// eslint-disable-next-line import/no-deprecated -- ContainerRuntime class to be moved to internal scope
+	public async instantiateFirstTime(runtime: ContainerRuntime): Promise<void>;
+	/**
+	 * Called the one time the container is created, and not on any subsequent load.
+	 * i.e. only when it's initialized on the client that first created it
+	 * @param runtime - The runtime for the container being initialized
+	 */
+	public async instantiateFirstTime(runtime: IContainerRuntime): Promise<void>;
+	public async instantiateFirstTime(
+		// eslint-disable-next-line import/no-deprecated -- ContainerRuntime class to be moved to internal scope
+		runtime: IContainerRuntime | ContainerRuntime,
+	): Promise<void> {
 		await this.containerInitializingFirstTime(runtime);
 		await this.containerHasInitialized(runtime);
 	}
 
-	public async instantiateFromExisting(runtime: ContainerRuntime): Promise<void> {
+	/**
+	 * @deprecated Use the overload taking in an IContainerRuntime
+	 */
+	// eslint-disable-next-line import/no-deprecated -- ContainerRuntime class to be moved to internal scope
+	public async instantiateFromExisting(runtime: ContainerRuntime): Promise<void>;
+	/**
+	 * Called every time the container runtime is loaded for an existing container.
+	 * i.e. every time it's initialized _except_ for when it is first created
+	 * @param runtime - The runtime for the container being initialized
+	 */
+	public async instantiateFromExisting(runtime: IContainerRuntime): Promise<void>;
+	public async instantiateFromExisting(
+		// eslint-disable-next-line import/no-deprecated -- ContainerRuntime class to be moved to internal scope
+		runtime: IContainerRuntime | ContainerRuntime,
+	): Promise<void> {
 		await this.containerHasInitialized(runtime);
 	}
 
+	/**
+	 * @deprecated Use the overload that returns an IContainerRuntime & IRuntime
+	 */
+	public async preInitialize(
+		context: IContainerContext,
+		existing: boolean,
+		// eslint-disable-next-line import/no-deprecated -- ContainerRuntime class to be moved to internal scope
+	): Promise<ContainerRuntime>;
+	/**
+	 * Called at the start of initializing a container, to create the container runtime instance.
+	 * @param context - The context for the container being initialized
+	 * @param existing - Whether the container already exists and is being loaded (else it's being created new just now)
+	 */
+	public async preInitialize(
+		context: IContainerContext,
+		existing: boolean,
+	): Promise<IContainerRuntime & IRuntime>;
```

Comment:
I added `& IRuntime` here to match the type specified on the base class `RuntimeFactoryHelper`